src/acceleration/Acceleration.cpp

```diff
@@ -40,10 +40,8 @@ void Acceleration::applyRelaxation(double omega, DataMap &cplData, double window
         auto  oldGradients = couplingData.getPreviousGradientsAtTime(stample.timestamp); // IMPORTANT DETAIL: The interpolation that we use for resampling does not necessarily have to be the same interpolation as the interpolation the user accesses via read-data. (But probably it is easier to just use the same)
         gradients          = gradients * omega + oldGradients * (1.0 - omega);
       }
+      couplingData.setSampleAtTime(stample.timestamp, stample.sample); // @todo try to remove, currently needed to update CouplingData::_data::_sample. Should not be needed anymore when https://github.com/precice/precice/issues/1645 is resolved.
```

Comment:
We could also call this function outside of the loop for the time associated with the last stample. I think it is (a bit) cleaner this way. In any case it is a (hopefully temporary) hack..

Reply:
Would it make sense to make this explicit with a `setGlobalSample()` until we can get rid of it?

Reply:
Done.